models/DataObject/ClassDefinition/Data/Multiselect.php

```diff
@@ -105,18 +113,36 @@ public function getRenderType(): ?string
         return $this->renderType;
     }
 
+    public function setDefaultValue(array $defaultValue): static
+    {
+        $this->defaultValue = $defaultValue;
+
+        return $this;
+    }
+
+    public function getDefaultValue(): ?array
+    {
+        return $this->defaultValue;
+    }
+
     /**
      * @see ResourcePersistenceAwareInterface::getDataForResource
      *
      *
      */
     public function getDataForResource(mixed $data, DataObject\Concrete $object = null, array $params = []): ?string
     {
-        if (is_array($data)) {
+        if (!$this->isEmpty($data) && is_array($data)) {
             return implode(',', $data);
         }
 
-        return null;
+        $defaultValue = $this->handleDefaultValue($data, $object, $params);
+
+        if (is_array($defaultValue)) {
+            return implode(',', array_map(fn ($v) => $v['value'] ?? $v, $defaultValue));
+        }
+
+        return $defaultValue ?? '';
```

Comment:
```suggestion
        return $defaultValue;
```
was already nullable, should be fine